.github/workflows/build_wheels.yml

```diff
@@ -1,9 +1,6 @@
 name: Build wheels
 
 on:
-  # Trigger on pull requests.
-  pull_request:
```

Comment:
Can we continue to use cibuildwheel to build/test macOS and Linux, and only use the `run_tests.yml` script for Windows? I strongly prefer testing with cibuildwheel wherever possible because it ensures consistency in our building/testing process between releases and normal commits. That reduces the risk of build problems emerging at release time.

To do that, we'd need to re-enable `pull_request` triggers for `build_wheels.yml` and disable `macOS-latest, ubuntu-latest` runners on `run_tests.yml`.

Reply:
That's a fair concern. The reason I originally only wanted to use `run_tests.yml` for regular commits was so we don't have to build tbb from source just to unit test. The tradeoff is CI time vs the release process working security.

I don't believe hoomd runs cibuildwheel on every commit, so maybe @joaander can weigh in on this idea before we decide how to move forward.

Reply:
I prefer the simplicity of the single action that tests on all platforms. This is easier to update and keep consistent. As Tommy mentioned, it also reduces the CI times run on every commit and tests that users can build and install freud against conda-forge prereqs. `cibuidwheel` tests a very narrow environment matching a specific set of old OSes and specific dependencies provided by `pip`.

I'm in favor of deferring the `cibuildwheel` tests to limited runs at release time, or run once just before merging a PR. In `gsd` I do run the `cibuildwheel` builds AND the action based builds on every commit, but gsd is very quick to build and test.

Reply:
I haven't looked at the CI run times but that seems like a good argument for not using cibuildwheel if it's substantially different. I agree that being able to test against conda-forge dependencies is helpful. My preference for consistent CI environments between normal commits and releases is partly driven by the headaches I faced in making releases of the package in the past. Things are much more stable and better tested now, so do whatever you think is best here. I'd say running cibuildwheel on PR merge is a good compromise. The CI build time on the main branch is usually not a blocker, and it ensures that tests are performed regularly and problems can be caught before release.